docs/api/proposed_endpoints.md

```diff
@@ -0,0 +1,186 @@
+[TOC]
+
+# Posts related actions
+
+## Create a post
+
+Create a post of contents.
+
+```
+POST api/posts
```

Comment:
If we don't require Authorization in order to create a Post then it'll be difficult to set `posted_by` and also Update endpoint to authenticate the original author of the post.
Hence, we should be using `(requires authentication)` for creating the post as well.

Reply:
Authentication is done using headers. You don't need to add another endpoint for that.

Reply:
It is just indicating that this particular API endpoint will only allow authenticated user.

The authentication is done in this API only. After login, the user gets a `token` that they will provide in subsequent requests to indicate who they really are. From that token, we know who the requester is and can then check whatever permissions are necessary.

Does that make sense, @storymode7 @GeekyShacklebolt